projects/storefrontlib/src/cms-components/checkout/services/checkout-config.service.ts

```diff
@@ -55,6 +61,51 @@ export class CheckoutConfigService {
     return stepIndex >= 0 ? stepIndex : null;
   }
 
+  protected sortDeliveryModes(deliveryMode1, deliveryMode2) {
+    if (deliveryMode1.deliveryCost > deliveryMode2.deliveryCost) {
+      return 1;
+    } else if (deliveryMode1.deliveryCost < deliveryMode2.deliveryCost) {
+      return -1;
+    } else {
+      return 0;
+    }
+  }
+
+  protected getFirstCodeOrGoToNextMode(deliveryModes: DeliveryMode[], index) {
+    const lastMode = this.defaultDeliveryMode.length - 1 === index;
+    return lastMode
+      ? deliveryModes[0].code
+      : this.findMatchingDeliveryMode(deliveryModes, index + 1);
+  }
+
+  protected findMatchingDeliveryMode(deliveryModes: DeliveryMode[], index = 0) {
+    switch (this.defaultDeliveryMode[index]) {
```

Comment:
I would create here variable for storing `defaultDeliveryModeCode`. In each case when condition matches assign to variable and break. Only at the end of the function if this variable is still undefined/null invoke `getFirstCodeOrGoToNextMode` function.